test/test.js

```diff
@@ -728,3 +729,75 @@ describe('pipeline execution', () => {
     });
   }
 });
+
+describe('function execution', () => {
+  for (const envConfig of envConfigs) {
+    describe(envConfig.name, () => {
+      beforeAll(envConfig.setup);
+      afterAll(envConfig.teardown);
+      describe(envConfig.name + ' extended json decode (incoming)', () => {
+        beforeAll(() => {
+          fetchMock.restore();
+          fetchMock.post(LOCALAUTH_URL, {userId: '5899445b275d3ebe8f2ab8a6'});
+          fetchMock.post(FUNCTION_URL, (name, arg1, arg2) => {
+            return JSON.stringify({result: [{x: {'$oid': hexStr}}]});
+          });
+        });
+
+        it('should decode extended json from function responses', () => {
+          expect.assertions(1);
+          let testClient = new StitchClient('testapp');
+          return testClient.login('user', 'password')
+            .then(() => testClient.executeFunction('testfunc', {items: [{x: {'oid': hexStr}}]}, 'hello'))
+            .then((response) => expect(response.result[0].x).toEqual(new ejson.bson.ObjectID(hexStr)));
+        });
+      });
+
+      describe('extended json encode (outgoing)', () => {
+        let requestArg;
+        beforeAll(() => {
+          fetchMock.restore();
+          fetchMock.post(LOCALAUTH_URL, {userId: hexStr});
+          fetchMock.post(FUNCTION_URL, (name, arg1, arg2) => {
```

Comment:
it looks like we don't use `name, arg1, arg2` at all in the function. do we need them? i'm assuming the intention was to assert that the server receives args of the same value that matches what was provided to the client. But I think we're already doing that with the assertion on L773.